sites/next.skeleton.dev/src/pages/docs/[...slug]/index.astro

```diff
@@ -16,22 +15,17 @@ export const getStaticPaths = (async () => {
 }) satisfies GetStaticPaths;
 
 const { entry } = Astro.props;
-const { Content, remarkPluginFrontmatter} = await entry.render();
+const { Content, remarkPluginFrontmatter } = await entry.render();
 
-// Split the route into segments
-const routeArr = entry.slug.split('/');
-
-// Create allowlist for sections that use split meta/framework format
-const metaCollections = [ 'components' ];
-const meta = metaCollections.includes(routeArr[0]) ? (await getEntry('docs', `${routeArr[0]}/${routeArr[1]}/meta`)) : null;
+// Get meta for the current page
+// This Regex replaces that last part of the url with meta, so "components/accordion/svelte" becomes "components/accordion/meta"
+const meta = await getEntryBySlug('docs', entry.slug.replace(/\/[^\/]*$/, '/meta'));
```

Comment:
As discussed on Discord, you've generalized this to merge `meta.mdx` if it exists. This unfortunately is a bit of regression currently because it's merging:

`/content/docs/get-started/meta.mdx` over
`/content/docs/get-started/installation.mdx`

This results in missing or wrong frontmatter for most sections. The title, description, order, etc are all messed up.

I'd suggest one of two changes:

1. Keep your new script logic, but delete the `meta.mdx` from locations they are not used (anything but `/components`)
2. Revert to the previous logic, which contains a whitelist of sections we allow meta merging to occur.

Reply:
For the record, we've opted for option 1